File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-java/src/main/java/com/example/statemachine/simple/doument/config/guard/PersistGuard.java
package com.example.statemachine.simple.doument.config.guard;

import com.example.statemachine.simple.doument.DocumentEvent;
import com.example.statemachine.simple.doument.DocumentState;
import com.example.statemachine.simple.doument.config.DocumentIdChecker;
import com.example.statemachine.simple.doument.config.exception.PersistenceDataIsNotExistException;
import java.util.Objects;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.statemachine.StateContext;
import org.springframework.statemachine.guard.Guard;

public class PersistGuard implements Guard<DocumentState, DocumentEvent> {

  private final DocumentIdChecker documentIdChecker;

  public PersistGuard(JdbcTemplate jdbcTemplate) {

    this.documentIdChecker = new DocumentIdChecker(jdbcTemplate);


  }

  @Override
  public boolean evaluate(StateContext<DocumentState, DocumentEvent> context) {

    final var messageHeaders = context.getMessageHeaders();

    if (!messageHeaders.containsKey("document")) {
      return false;
    }

    try {
      final var documentId = Objects.requireNonNull(
        messageHeaders.get("document", Integer.class));


      return documentIdChecker.isExist(documentId);
    } catch (NullPointerException | IllegalArgumentException | PersistenceDataIsNotExistException ex) {
      return false;
    }
  }
}

File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-kotlin/src/main/kotlin/com/example/statemachine/simple/document/config/guard/PersistGuard.kt
package com.example.statemachine.simple.document.config.guard

import com.example.statemachine.simple.document.DocumentEvent
import com.example.statemachine.simple.document.DocumentState
import com.example.statemachine.simple.document.config.DocumentIdChecker
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.statemachine.StateContext
import org.springframework.statemachine.guard.Guard

class PersistGuard(
  jdbcTemplate: JdbcTemplate
) : Guard<DocumentState, DocumentEvent> {

  private val documentIdChecker: DocumentIdChecker = DocumentIdChecker(jdbcTemplate)

  override fun evaluate(context: StateContext<DocumentState, DocumentEvent>): Boolean {
    val messageHeaders = context.messageHeaders

    if (!messageHeaders.containsKey("document")) {
      return false
    }

    val documentId = messageHeaders.get("document", java.lang.Integer::class.java)!!.toInt()

    return documentIdChecker.isExist(documentId)
  }
}
